### ISSUE TITLE
Allow poll owners to edit the names of already existing options

### ISSUE THREAD

**Issue by LuHa9** [nextcloud/polls#2463]
## Describe the goal you'd like to achieve
As a poll owner, I want to be able to edit the names of options that already exists in text-based polls, because it avoids that all poll users have to remake their choices after a tiny change in one option. In the current state I have to delete and recreate the option with the new name, so all poll users have to remake their choices.

**Comment by dartcafe** [comment]
If you change the text, you will invalidate all votes, because the votes rely on the entered text as the identifier. So sorry, but no.

I understand your intention, but what you describe a as value has to be avoided. If you have votes opting in for an option and the meaning of the option gets changed (from good to bad), the votes don't represent the voters optinion anymore. For that reason the votes get invalidated, if the text changes. 

To avoid confusion and manipulation the easy way, texts are not changeable. 

Also see  #1696 and #1079 with similair request.

**Comment by LuHa9** [comment]
I do also understand your intention but I personally don't see the problem. Every edit could be made transparently (activity log, notification to the poll users) so the poll users will be aware of those edits. But this is not the place to discuss such policy issues;-)

**Comment by dartcafe** [comment]
I agree, it is solvable but it will result in a lot of work (coding, testing, side effects), more complexity and a lot of new notifications or warnings in the UI. And possibly a tracking of changes is necessary.

**Comment by LnLcFlx** [comment]
I think this can be achieved with relatively minor changes (correct me if I am wrong please).
It is only necessary to swap the ´vote_option_text´ as an identifier for the ´vote_option_id´ (which is already stored in ´oc_polls_votes´).

And to address the issue of manipulating the meaning of votes, one could add a little warning icon above the ticks/crosses/(ticks)  of participants if the text that was voted on does not match the current text.
And hovering over that icon could show what text was actually voted on.
For anonymous polls this is a little more difficult but (for the time being) one could disable editing for anonymous polls.

In the long run tracking of option changes as suggested above would be great though.

**Comment by dartcafe** [comment]
I will repost my last statement: 

> I agree, it is solvable but it will result in a lot of work (coding, testing, side effects), more complexity and a lot of new notifications or warnings in the UI. And possibly tracking of changes will be necessary.

When you want to change the text of an option: Create a new one and delete the invalid option. It's done by only entering a text and two mouse clicks. Changing the text is the same effort. Two clicks and changing the option text. What is the value comparing to the costs of designing, coding and testing this change?
